---
title: "Final Report: Relationship between Microbiome Composition and Development of Breast Cancer"
author: "Charlotte Lopp"
date: "December 2, 2017"
output:
    github_document: default
    word_document: default
bibliography: references.bib
csl: bioinformatics.csl
---

# Introduction 

Breast cancer affects affects one in eight women in their lifetime and according to the American Cancer Society, it is the second leading cause of cancer death in women with only lung cancer killing more women each year. Over the past twenty years $5.5 billion has been dedicated to breast cancer research but the origins of a majority of breast cancer cases remain unknown [@madigan1995proportion]. Part of what is known is that migration studies have shown an increased incidence of breast cancer among migrants and their descendants after they move from a region of low breast cancer risk to a region of high risk [@le2002cancer, @shimizu1991cancers]. Environmental factors as well as genetics play a role in disease development but which environmental factors and the extent of their influence is still being explored [@urbaniak2016microbiota]; diet, age and genetic disposition are established risk factors [@xuan2014microbial]. In fact, there are many epidemiological evidences and research studies that have suggested that diet plays an important role in breast cancer prevention and progression [@aragon2014modification]. Specifically, there are recent studies that suggest that eating fermented foods such as probiotic yogurt play a role in providing protection [@lakritz2014beneficial]. The recent appreciation of the influential role that microbiota has on human health and disease has led to the investigation of if microbes play a role in breast cancers and if they do, to what extent [@xuan2014microbial].

The human microbiota refers to the collection of microbes inhabiting the human body; microbes inhabiting the human body outnumber human cells 10:1 [@xuan2014microbial]. This microbiota plays an integral role in human development and changes in an individual's composition have or an imbalance has been implicated in various human diseases. Individuals with periodonitis [@darveau2010periodontitis, @ximenez2000comparison], inflammatory bowel disease [@frank2007molecular], psoriasis [@gao2008substantial], asthma [@hilty2010disordered], diabetes [@larsen2010gut], bacterial vaginosis [@hummelen2010deep], and colorectal cancer [@mira2015microbial] have different bacterial communities than individuals. Although it is unclear whether the these microbial differences are a consequence or a cause of the disease, there is evidence in favor of a causal relationship, as healthy animals transplanted with feces from those with obesity, colitis, or colorectal cancer then go on to develop disease [@garrett2007communicable, @turnbaugh2009core, @zackular2013gut]. In obese individuals, the ratio of Firmicutes to Bacteroidetes in the colon is significantly higher than in lean individuals [@turnbaugh2006obesity, @ley2006microbial]. Placing these obese individuls on low-fat diets resulted in a decrease in this ratio, although not to the levels seen in lean individuals [@ley2006microbial]. In colon cancer, the overabundance of a single bacterial species *Fusobacterium nucleatum* correlates with disease and increased likelihood of lymph node metastasis [@castellarin2012fusobacterium]. In contrast to the detrimental affects of *Fusobacterium nucleatum*, the bacterium *Bacteroidetes fragilis* exerts a protective effect against colitis by modulating inflammatory immune responses in the gut [@mazmanian2008microbial]. These recent studies have made it increasingly apparent that microbiota composition as well as specific bacterial species can have either preventive or progressive effects in terms of maintaining health or encouraging disease development, respectively [@xuan2014microbial]. Thus, we are led to the question of what bacterial species, if any, have effect on breast cancer and how are these effects are manifested.  

The first step in answering this question is determining that there is in fact a microbiome within the mammary tissue - a question that was addressed by Urbanick et al. (2014). In the female mammary gland, milk has been shown to contain bacterial species which ostensibly reach the ducts from the skin [@urbaniak2014microbiota]. The rationale of Urbaniak et al. (2014) was that given the nutrient-rich fatty composition of the female breast, the widespread vasculature and lympathics, and the diffuse location of the lobules and ducts leading from the nipple, bacteria would be widespread within the mammary glands, irrespective of lactation. Using 16S rRNA sequencing and culture, they analyzed breast tissue from 81 women with and without cancer in Canada and Ireland. A diverse bacteria population was detected within the tissues collected from sites all around the breast in women raging from 18 to 90 years old, not all of whom had a history of lactation. They stated that they successfully confirmed a breast microbiome in this study and added the study by Xuan et al. [@xuan2014microbial] that established the same conclusion. They found that *Proteobacteria* was the most abundant phylum in breast tissue where members of this phylum make up only a small portion of the overall bacterial community - a case unlike those seen in the vagina, oral cavity, bladder, skin, and gastrointestinal tract. This finding was consistent with other pieces of literature and suggested that breast tissue may in fact have a unique microbiota, distinct from that found at other body sites. The higher abundance of *Proteobacteria* and *Firmicutes* (specifically the class *Bacilli*) was perhaps due to host microbial adaptation to the fatty acid environment in the tissue. Another thing of interest was that *Proteobacteria* is also the principle phylum in human milk [@ward2013human] along with many of the same bacteria that they had detected in the tissue which raises the possibility that the tissue microbiota could be a source of bacterial inocula for babies. Despite having some difficulties with culturing different bacterial strains, all the bacteria that were cultured were also detected by 16s rRNA sequencing, supporting the argument that the DNA amplimers were from viable bcateria and not solely from remnant bcaterial DNA [@urbaniak2014microbiota].  

The use of 16S rRNA gene sequences for bacterial identification has been by far the most common housekeeeping genetic marker used for a multitude of reasons [@janda200716s]. The highest ranking of these include: (1) its presence in almost all bacteria, often existing as a multigene family, or operons; (2) the function of the 16S rRNA gene over time has not changed, suggesting that random sequence changes are a more accurate measure of time (evolution); and (3) the 16S rRNA gene (1,500 bp) is large enough for informatics purposes [@patel200116s]. 16S rRNA gene sequence analysis can better identify poorly described, rarely isolated, or phenotypically aberrant strains, can be routinely used for identification of mycobacteria, and can lead to the recognition of novel pathogens and noncultured bacteria [@clarridge2004impact]. It is easy to use and performs well as well in comparison to the more cumbersome manipulations of involving the "gold standard" for proposed new species DNA-DNA hybridization investigations [@janda200716s]. One of the most attractive aspects of 16S rRNA gene sequence informatics is to provide genus and species identification for isolates that do not fit any recognized biochemical profiles, for strains generating only a "low likelihood" or "acceptable" identification according to commercial systems, or for taxa that are rarely associated with human infectious diseases [@janda200716s]. The Illumina MiSeq platform (San Diego, CA, USA) provides researchers with a scalable, high-throughput and streamlined sequencing platform to survey community composition from clinical and environmental samples [@fadrosh2014improved]. Thus, Urbaniak et al. (2016) used this method to identify taxa that may affect breast cancer in the relatively newly established breast microbiome.  

As they had already established that breast tissue microbiome exists in Canadian and Irish women [@urbaniak2014microbiota], Urbaniak et al. then went on to determine whether this local microbiome plays a role in modulating the risk of breast cancer development [@urbaniak2016microbiota]. From a new patient study group, they examined the breast microbiota of 70 women who varied across having breast cancer (normal adjacent tissue collected) or benign tumors (normal adjacent tissue collected) or were disease free. They isolated bacteria from cancer patients and then characterized and examined them for their abilities to induce DNA damage. The endeavour of Urbaniak et al. was to differentiate between baterial profiles in breast tissue of healthy women and those with breast cancer and see which bacteria influenced the progression or prevention of breast cancer. 

# Methods

## Study design

Urbaniak et al. collected fresh breast tissue from 71 women ranging from 19 to 90 years old who were undergoing breast surgery at St. Joseph's Hospital in London, Ontario, Canada. 58 of these women underwent lumpectomies or mastectomes for neither benign (*n* = 13) or cancerous (*n* = 45) tumors, and 23 were free of disease and underwent either breast reductions or enhancements. For those women with tumors, the tissue obtained for analysis was collected outside the marginal zone, approximately 5 cm away from the tumor.

## Sample origin and sequencing

After excision, fresh tissue was immediately placed in a sterile vial on ice and homogenized within 30 minutes of collection. After tissue homogenates in sealed containers were thawed on ice, procedures for DNA isolation ensued. For the 16S rRNA gene sequencing, PCR amplification, the genomic DNA isolated from the clinical samples was amplified using barcoded primers that amplified the V6 hypervariable region of the 16S rRNA gene (70 bp long): V6-forward, 5′ACACTCTTTCCCTACACGACGCTCTTCCGATCTnnnn(8)CWACGCGARGAACCTTACC3′; and V6-reverse, 5′CGGTCTCGGCATTCCTGCTGAACCGCTCTTCCGATCTnnnn(8)ACRACACGAGCTGACGAC3′. The pooled PCR purified sample was then paired-end sequenced on the Illumina Mi-Seq platform using a 150 cycle kit with a paired-end 80-bp run at the London Regional Genomics Center, London, Ontario, Canada, following standard operating procedures. Custom Perl [@lindner2004easylinkage] and Bash [@badal2006package] scripts were used to to demultiplex the reads and assign barcoded reads to individual samples followed by multiple layers of filtering. 

## Computational

Computation with this data began with creating a bashscript with code which downloaded the list of 68 files (from the NCBI Sequence Read Archive study number SRP076038) in the run table to the project's data directory. Before we could do anything else, we have to load the following libraries: mctoolsr [@leff2015mctoolsr], seqinr [@charif2016seqinr], citr [@wickham2015devtools], dplyr [@wickham2015dplyr], tidyr [@wickham2014tidyr], knitr [@xie2013knitr], dada2 [@callahan2016], ggplot2 [@wickham2015ggplot2] and phyloseq [@mcmurdie2013]. After looking at the quality profiles of all 68 samples, we adjusted the filter and trim parameters to a maximum length of 100 and a minimum length of 55. We then created a table of all the samples of their original lengths and then their output lengths after trimming to ensure that everything looked correct. The next steps was plotting the error models we built to ensure that the error models match the data. Next, we got rid of any duplicate sequences and then ran DADA2. Then, we created a species matrix ("sequence_table"), made a histogram in order to look at the distribution of the trimmed and denoised sequences, and removed any chimeras. We proceeded to make a table of the outputs of each step ("track"), assigned taxonomy ("taxa"), and then used this taxonomy to create a phylogenetic tree with Geneious [@drummond2011geneious]. Finally, we created a phyloseq object from DADA2 outputs and saved everything to a file ("output/phyloseq_obj.RData") which was then loaded into the Rmd file for the final report using the 'load()' function. 

# Results

```{r load-libraries, message = FALSE, echo = FALSE}
# Be sure to install these packages before running this script
# They can be installed either with the intall.packages() function
# or with the 'Packages' pane in RStudio

# load general-use packages
library("dplyr")
library("tidyr")
library("knitr")
library("ggplot2")

# this package allows for the easy inclusion of literature citations in our Rmd
# more info here: https://github.com/crsh/citr
# and here:
# http://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html
library("citr")

# These are the primary packages well use to clean and analyze the data
# this package needs to be installed from bioconductor -- it's not on CRAN
# see info here: https://benjjneb.github.io/dada2/dada-installation.html
library("dada2")

# This to export a fasta of our final denoised sequence variants
library("seqinr")

# To install this you have to install from GitHub
# See more info here: https://github.com/leffj/mctoolsr
# run this -- install.packages("devtools")
# and then this -- devtools::install_github("leffj/mctoolsr")
library("mctoolsr")

# And this to visualize our results
# it also needs to be installed from bioconductor
library("phyloseq")

# Add this for combining two different graphs
library("grid")

# This loads my separate R script file that processed the data
load("output/phyloseq_obj.RData")
```

In order to look at bacterial relative abundance in a meaningful way, we had to first narrow down the number of genera we were looking at by separating those with the highest abundance. Figure 1 shows the 11 most abundant genera seen in normal adjacent tissue in patients with benign tumors, normal adjacent tissue in patients with cancerous tumors, and normal tissue in healthy patients. The relative abundance seen in patients wih benign tumors, from greatest to least, is *Micrococcus*, *Bacilus*, *Pseudomonas*, *Escherichia/Shigella*, *Acinetobacter*, *Propionibacterium*, and *Delftia*. Looking at patients with cancerous tumours, we see the following relative abundance: *Pseudomonas*, *Bacilus*, *Escherichia/Shigella*, *Delftia*, *Propionibacterium*, *Staphylococcus*, and *Acinetobacter*. The relative abundance in healthy patients is as follows: *Pseudomonas*, *Micrococcus*, *Prevotella*, *Escherichia/Shigella*, *Propionibacterium*, *Acinetobacter*, and *Staphylococcus*. The bacterial abundance in patients with benign tumors is two-fold of that which is seen in both patients with cancerous tumors and healthy patients.

```{r bar-plot-relative-abundance, echo = FALSE}
melted_phyloseq <- psmelt(phyloseq_obj)

top_50 <- melted_phyloseq %>%
  group_by(OTU, Abundance) %>%
  summarize(mean_abund = mean(Abundance)) %>%
  arrange(desc(mean_abund)) %>%
  head(n = 50)

top_genes <- top_50 %>%
  ungroup() %>%
  select(OTU) %>%
  unique() %>%
  pull()

#df %>% group_by(path1) %>% mutate(relAbundByPath = value / sum(value))
 
melted_phyloseq %>%
  filter(OTU %in% top_genes) %>%
  group_by(Genus, tissue_type, tissue_type.1) %>%
  summarize(mean_abund = mean(Abundance)) %>%
  ggplot(aes(x = tissue_type,
             y = mean_abund,
             fill = Genus)) +
    geom_col(color = "black") +
  facet_wrap( ~ tissue_type.1) +
    xlab("Tissue Type") +
    ylab("Relative Abundance") +
    ggtitle("Relative Abundance of Bacterial Genera Between Tissue Types")
```

**Figure 1**: This figure shows the relative abundance of bacterial genera among tissue types. 

While Figure 1 allows for visualization of the bacterial abundance relative to one another within a tissue type, the relative abundance of a bacterial genus across all three tissue is not as easily discernable. Thus, our next step was to calculate the mean and standard deviation of each bacterial abundance across all three tissue types for comparison amongst one another (Figure 2). A higher abundance is seen in cancer patients than healthy patients for *Acinetobacter*, *Bacillus*, *Delftia*, *Escherichia/Shigella*, *Micrococcus*, *Staphylococcus*, and *Propionibacterium* (although this difference is barely discernable). A higher abundance is seen in healthy patients than cancer patients for *Lactobacillus*, *Prevotella*, and *Ralstonia*. There is no noticeable difference in terms of abundance between cancer or healthy patients for *Pseudomonas*. The largest margins of difference in relative abundance between cancer patients and healthy patients are seen in *Bacillus*, *Escherichia/Shigella*, and *Micrococcus*. However, for *Micrococcus* benign tumor patients have the greatest relative abundance, then healthy patients have the next highest, followed by cancer tumor patients who have a relative abundance of zero.

```{r mean-and-sd-rel-abund, echo = FALSE}
melted_phyloseq %>%
  filter(OTU %in% top_genes) %>%
  group_by(Genus, tissue_type.1) %>%
  summarize(mean_abund = mean(Abundance),
            sd_abund = sd(Abundance)) %>%
  ggplot(aes(x = tissue_type.1,
             color = Genus)) +
    geom_errorbar(aes(ymin = mean_abund - sd_abund,
                      ymax = mean_abund + sd_abund),
                  width = 0.25) +
      geom_point(aes(y = mean_abund)) +
  facet_wrap( ~ Genus) + 
  theme(axis.text.x =
            element_text(angle = 90,
                         hjust = 1)) +
    xlab("Tissue Type") +
    ylab("Relative Abundance") +
    ggtitle("Mean and Standard Deviation of Relative Abundance of Bacterial 
            Genera Between Tissue Types Separated by Genus")
```

**Figure 2**: This figure shows the relative abundance via mean and standard deviation of bacterial genera between tissue types, futher separated by genus. 

We were then curious about the relationship between the relative abundance of bacterial genera and the type of tumor that the cancer patients had (Figure 3). *Pseudomonas* is seen in about the same amount throughout all of the types of tumors. However, for "phyllodes & fibroadenoma," "phyllodes tumor," "invasive mammary carcinoma," "invasive carcinoma," "DCIS (multifocal)," "invasive mammary carcinoma, tubular features," "scerlosed intraductal papilloma, focal atypia" and "intraductal papilloma" *Pseudomonas* is almost all (a very high majority) of the abundance for each. For "calcification" almost all of the abundance is *Bacillus* with the remainder being *Pseudomonas*, *Delftia*, and *Staphylococcus*. For "DCIS," the abundance is mostly *Bacillus* and *Pseudomonas* with the remainder being *Escherichia/Shigella* and *Propionibacterium*. For both "encapsulated pillary carcinoma" and "invasive lobular" the two most promiment bacterial genera are *Bacillus* and *Pseudomonas*. For "fibroadenoma" the highest abudance is seen in "Micrococcus" followed by *Acinetobacter*, *Bacillus*, *Escherichia/Shigella* and *Propionibacterium*. Almost all of "intraductal papillomata, no atypia" is "Micrococcus" with the only other discernable bacteria being *Pseudomonas*. For "invasive" a very high abundance of *Staphylococcus* is seen with the rest being *Pseudomonas*. A similar pattern to "fibroadenoma" is seen in "invasive ductal" in terms of the bacteria present; the abundance is (from greatest to least) *Escherichia/Shigella*, *Delftia*, *Acinetobacter*, *Bacillus*, *Propionibacterium*, and *Pseudomonas*. For "mammary hamartoma" the abundance is split between *Bacillus*, *Pseudomonas*, and *Delftia*. *Escherichia/Shigella* and *Pseudomonas* make up practically all of "papillary lesion without atypia" with a barely discernable presence of *Acinetobacter* and *Propionibacterium*. For "N/A," which is referring to breast reductions or breast augmentations since these were the healthy patients undergoing surgery, we see *Pseudomonas*, *Micrococcus*, *Prevotella*, *Propionibacterium*, *Lactobacillus*, *Escherichia/Shigella*, and *Acinetobacter*.

```{r type-of-tumour-by-abundance, echo = FALSE}
melted_phyloseq %>%
  filter(OTU %in% top_genes) %>%
  group_by(Genus, type_of_tumour) %>%
  summarize(mean_abund = mean(Abundance)) %>%
  ggplot(aes(x = type_of_tumour,
             y = mean_abund,
             fill = Genus)) +
    geom_col(color = "black") +
  theme(axis.text.x =
            element_text(angle = 90,
                         hjust = 1)) +
      ylab("Relative Abundance") +
      xlab("Type of Tumour") +
      ggtitle("Relative Abundance of Bacterial Genera Among Tumour Types")
```

**Figure 3**: This figure shows the relative abundance of bacterial genera among the different types of tumors seen in the cancer patients.

After seeing the relationship between relative abundance of bacterial genera and tumor type, we were interested to see what the relationship was between relative abundance of bacterial genera and cancer stage (Figure 4). Here we see that the highest abundance is seen in cancer stage 4 while the number of patients with stage 4 cancer was the lowest (one person); the relative abundance from highest to lowest is *Escherichia/Shigella*, *Pseudomonas*, *Propionibacterium*, *Bacillus* and *Acinetobacter*. Stage 3 cancer, which had the second lowest number of patients, is made up of primarily *Staphylococcus* and *Pseudomonas*. Stage 2 cancer, which had the same number of patients as stage 0 cancer, only had *Pseudomonas* as the discernable bacteria. For cancer stage 1, which had the most patients, the abundance is *Pseudomonas*, *Bacillus*, *Escherichia/Shigella* and *Delftia*. Finally, for stage 0 cancer, the abundance is *Pseudomonas*, *Bacillus*, and *Escherichia/Shigella*. 

```{r cancer-stage-by-abundance, echo = FALSE}
p1 <- sample_data(phyloseq_obj) %>%
  group_by(stage_of_cancer) %>%
  tally() %>%
  ggplot(aes(x = stage_of_cancer,
             y = n,
             fill = stage_of_cancer)) +
  geom_col(position = "dodge") +
  xlab("Cancer Stage") +
  ylab("Count") +
  ggtitle("Relative Abundance of Bacterial Genera and Patient Count by Cancer Stage")

p2 <- melted_phyloseq %>%
  filter(OTU %in% top_genes) %>%
  group_by(Genus, stage_of_cancer) %>%
  summarize(mean_abund = mean(Abundance)) %>%
  ggplot(aes(x = stage_of_cancer,
             y = mean_abund,
             fill = Genus)) +
  geom_col(color = "black") +
  theme(axis.text.x =
          element_text(angle = 90,
                       hjust = 1)) +
  ylab("Relative Abundance") +
  xlab("Cancer Stage")

grid.newpage()
grid.draw(rbind(ggplotGrob(p1), ggplotGrob(p2), size = "last"))
```

**Figure 4**: This figure shows the relative abundance of bacterial genera for each cancer stage. 

Once we saw the relationship between relative abundance of bacterial genera and cancer stage, we wondered how tumor types were distributed across the cancer stages. Thus, we look at the relationship between the two with the relative abundance of bacterial genera 

```{r cancer-stage-by-tumour-type, echo = FALSE}
sample_data(phyloseq_obj) %>%
  group_by(stage_of_cancer, type_of_tumour) %>%
  tally() %>%
  ggplot(aes(x = stage_of_cancer,
             y = n,
             fill = type_of_tumour)) +
    geom_col(color = "black") + 
  ylab("Count") + 
  xlab("Cancer Stage") +
  ggtitle("Cancer Stage by Type of Tumour")
```

**Figure 5**: 

```{r bar-plot-rel-abundance-pylum, echo = FALSE}
melted_phyloseq %>%
  group_by(Phylum) %>%
  summarize(mean_abund = mean(Abundance),
            sd_abund = sd(Abundance)) %>%
  ggplot(aes(x = Phylum,
             color = Phylum)) +
    geom_errorbar(aes(ymin = mean_abund - sd_abund,
                      ymax = mean_abund + sd_abund),
                  width = 0.25) +
      geom_point(aes(y = mean_abund)) +
    theme(axis.text.x =
            element_text(angle = 90,
                         hjust = 1)) +
    xlab("Bacterial Phyla") +
    ylab("Relative Abundance") +
    ggtitle("Relative Abundance of Bacterial Phyla")
```

**Figure 6**: This shows the relative abundance between

```{r count-by-age-box-plot, echo = FALSE}
p3 <- melted_phyloseq %>%
  group_by(tissue_type.1, age) %>%
  ggplot(aes(x = tissue_type.1,
             y = age,
             color = tissue_type.1)) +
      geom_boxplot(width = 0.5) +
    xlab("Tissue Type") +
    ylab("Age") +
    ggtitle("Difference of Age at Breast Surgery by Tissue Type
A")

p4 <- melted_phyloseq %>%
  group_by(tissue_type.1) %>%
  summarize(mean_age = mean(age),
            sd_age = sd(age)) %>%
  ggplot(aes(x = tissue_type.1,
             color = tissue_type.1)) +
    geom_errorbar(aes(ymin = mean_age - sd_age,
                      ymax = mean_age + sd_age),
                  width = 0.25) +
      geom_point(aes(y = mean_age)) +
  xlab("Tissue Type") +
  ylab("Age") + 
  ggtitle("B")

grid.newpage()
grid.draw(rbind(ggplotGrob(p3), ggplotGrob(p4), size = "last"))
```

**Figure 7**

In addition to a minimum of 5-10 figures/tables (and associated captions), you should include sufficient text in this section to describe what your findings were. Remember that in the results section you just describe what you found, but you don't interpret it - that happens in the discussion. 2-3 pages.

# Discussion

Add around 3-4 pages interpreting your results and considering future directions one might take in analyzing these data.

Representative because it was not random sampling - it was a small number of people as well as limited by those who happened to be in the hospital to get some sort of breast surgery. 

# Sources Cited
